sherpa/astro/xspec/src/_xspec.cc

```diff
@@ -158,6 +164,62 @@ static PyObject* get_abund( PyObject *self, PyObject *args )
 }
 
 
+// See also: get_abund
+//
+// It is simpler to have separate routines rather than to try to deal
+// with the multiple options in one routine.
+//
+static PyObject* get_abund_from_table( PyObject *self, PyObject *args )
+{
+
+  // This requires both the table and element name.
+  //
+  char* table = NULL;
+  char* element = NULL;
+  if ( !PyArg_ParseTuple( args, (char*)"ss", &table, &element ) )
+    return NULL;
+
+  // Get the specific abundance. Unfortunately getAbundance reports an
+  // error to stderr when an invalid element is used, so we need to
+  // hide this. However it does throw an error if the table is unknown.
+  //
+  std::ostream* errStream = IosHolder::errHolder();
+  std::ostringstream tmpStream;
+  IosHolder::setStreams(IosHolder::inHolder(),
+			IosHolder::outHolder(),
+			&tmpStream);
+
+  float abundVal = 0.0;
+  try {
+    abundVal = FunctionUtility::getAbundance(string(table),
+					     string(element));
+  } catch (FunctionUtility::NoInitializer&) {
+    return PyErr_Format( PyExc_ValueError,
+			 "Unknown abundance table '%s'",
+			 table );
+  }
+
+  IosHolder::setStreams(IosHolder::inHolder(),
+			IosHolder::outHolder(),
+			errStream);
+
+  // Was there an error?
+  //
+  if( tmpStream.str().size() > 0 ) {
```

Comment:
When checking if strings contain a value "str.empty()" is typically preferred to "str.size()" for readability. Since the old code checked size(), no need to change it.

Reply:
I realized that one of the error conditions did not correctly restore the `IosHolder::setStreams` values so I added that. As I was doing that I also changed this (and the one other case) to say `if ( !tmpStream.str().empty() ) {`.